### DISCUSSION TITLE
Suggest installing image/warden updates

### DISCUSSION THREAD

**Discussion by ihor-sviziev** [wardenenv/warden#687] [Ideas]
Hi,
Pretty often I see the picture when people not updating their images for months, which means they use outdated versions of software, where some bugs or security issues not fixed. Same thing with warden itself, people not frequently updating it.
I suggest add some functionality that will automatically check the updates and suggest when it's available. I think the most relevant place would be running `warden env up`.

Examples:
- fetch the information about the used environment (and maybe service) images and if it's older than let's say 1 month - suggest running `warden env pull && warden env up` / `warden svc pull && warden svc up`.
- fetch information about latest warden version and suggest upgrade command when it's available.

On the other hand, we shouldn't force people to install updates, if they don't want to. So we should have an ability to disable this feature.
Also, I don't want to have some performance degradation all the time, so maybe we can use some file to check updates not earlier than once a month (or 2 weeks).

**Comment by bap14** [comment]
I like this idea. For keeping the images themselves up-to-date, adding a [Watchtower](https://containrrr.dev/watchtower) service container in monitor-only mode (or potentially make the mode configurable to allow auto-updating) would be the simple way to go. This should cover the service images (traefik, mailhog, etc.) as well as per-environment images (php, nginx, etc.).

Having Warden notify on available updates would be interesting. Could be as simple as adding an `.update-check` file to the warden home path, then add a configurable update check period (default it to 30? days), and for every invocation of `warden` just do a quick `mtime` check of the `.update-check` file and if it's beyond the configured period fetch the latest version information (from brew, or put latest version info on warden.dev, or just use GitHub API) and compare, show the notice (or not), and `touch` the `.update-check` file to adjust its `mtime`.

**Reply by navarr** [reply]
Right now, images are rebuilt every day as we don't have a good system to check for OS and PHP updates, and I believe due to that, the image checksum will constantly be different.  That means the images are _always_ out of date.  In order to properly prompt for image updates, we'd have to only update the images when there are actual updates.

Warden updates would be much more interesting and doable, I feel.

**Reply by ihor-sviziev** [reply]
@navarr what if we'll check if images are outdated for 30+ days?
